### PR TITLE
add DDoS protection

### LINKED ISSUE #43: DDoS protection
There has been a (presumably intentional) attack recently making circular requests:

```
2015-03-04T02:19:30.591826+00:00 heroku[router]: at=error code=H12 desc="Request timeout" method=GET path="/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://mustachify.me/?src=http://iscale.iheart.com/img/b64/ZGVmYXVsdA==?ops=max(1440,360)" host=mustachify.me request_id=3f156115-3e7d-45ad-bc6a-ba4b3ecbad90 fwd="54.87.226.46,173.245.54.42" dyno=web.2 connect=1001ms service=30001ms status=503 bytes=0
```

Non-mutually-exclusive Ideas:

* Tweak [timeout settings](http://www.rubydoc.info/github/jnunemaker/httparty/HTTParty/ClassMethods)
    * [Image fetching in magickly](https://github.com/afeld/magickly/blob/235374f1a8bcffb66a343187f1c6ce37d5f9a033/lib/dragonfly/data_storage/remote_data_store.rb#L15)
* Block `src` with a host of `mustachify.me`
* Add [rack-attack](https://github.com/kickstarter/rack-attack) (or similar)

Would love to hear about other creative ways to block these and other types of attacks.

### PR DESCRIPTION
**Pull request by afeld**
Closes #43.

### CODE REVIEW

**Comment by afeld** [comment]
Have the branch deployed... looking good so far!

screen shot 2015-03-04 at 6 54 57 am

**Comment by louim** [comment]
Excellent job!

File: lib/mustachio/rack_attack.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'active_support/cache'
+require 'active_support/cache/memory_store'
+
+class Rack::Attack
+  RPM = 10
```
Comment (afeld) [inline lib/mustachio/rack_attack.rb]: This number was picked fairly arbitrarily.

File: lib/mustachio/rack_attack.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  self.cache.store = ActiveSupport::Cache::MemoryStore.new
```
Comment (afeld) [inline lib/mustachio/rack_attack.rb]: Could move to a shared cache down the road, but this is good enough for now, and doesn't require any additional setup.

**Comment by parkr** [comment]
So great! :clap:

**Pull request merged**